osmdroid-shape/src/main/java/org/osmdroid/shape/ShapeConverter.java

```diff
@@ -83,74 +91,74 @@
 
+                            polygon.getFillPaint().set(shapeMetaSetter.getFillPaint());
+                            polygon.getOutlinePaint().set(shapeMetaSetter.getStrokePaint());
+                            folder.add(polygon);
                         }
                     }
-
                     break;
+
                     case POLYLINE: {
                         PolylineShape polylineShape = (PolylineShape) s;
                         for (int i = 0; i < polylineShape.getNumberOfParts(); i++) {
                             Polyline line = new Polyline(map);
-                            if (metadata != null) {
-                                metadata.setStringCharset(Charset.defaultCharset());
-                                line.setSnippet(metadata.toMap().toString());
-                                line.setTitle(getSensibleTitle(line.getSnippet()));
-                            }
+
                             PointData[] points = polylineShape.getPointsOfPart(i);
                             List<GeoPoint> pts = new ArrayList<>();
 
                             for (PointData p : points) {
-                                GeoPoint pt = new GeoPoint(p.getY(), p.getX());
+                                GeoPoint pt = fixOutOfRange(new GeoPoint(p.getY(), p.getX()));
                                 pts.add(pt);
                             }
-                            line.setPoints(pts);
+
+                            line.setPoints(pts);//points out of range should be fixed before this line.
+
+                            shapeMetaSetter.set(metadata, line);
+                            line.setSnippet(shapeMetaSetter.getSnippet());
+                            line.setTitle(shapeMetaSetter.getTitle());
+
+                            line.setSubDescription(shapeMetaSetter.getSubDescription());
+
+                            line.getOutlinePaint().set(shapeMetaSetter.getStrokePaint());
                             folder.add(line);
                         }
-
                     }
                     break;
+
                     case MULTIPOINT: {
                         MultiPointPlainShape aPoint = (MultiPointPlainShape) s;
 
                         PointData[] points = aPoint.getPoints();
                         for (PointData p : points) {
                             Marker m = new Marker(map);
-                            m.setPosition(new GeoPoint(p.getY(), p.getX()));
-                            if (metadata != null) {
-                                metadata.setStringCharset(Charset.defaultCharset());
-                                m.setSnippet(metadata.toMap().toString());
-                                m.setTitle(getSensibleTitle(m.getSnippet()));
+                            m.setPosition(fixOutOfRange(new GeoPoint(p.getY(), p.getX())));
 
-
-                            }
+                            shapeMetaSetter.set(metadata, m);
```

Comment:
Just `shapeMetaSetter.set(metadata, m)`: the other two lines `m.setSnippet(shapeMetaSetter.getSnippet())` and `m.setTitle(shapeMetaSetter.getTitle())` will be performed inside `shapeMetaSetter.set(metadata, m)`.